src/python/pants/option/options.py

```diff
@@ -397,11 +370,7 @@ def _make_parse_args_request(
 
     # TODO: Eagerly precompute backing data for this?
     @memoized_method
-    def for_scope(
-        self,
-        scope: str,
-        inherit_from_enclosing_scope: bool = True,
-    ) -> OptionValueContainer:
+    def for_scope(self, scope: str, check_deprecations: bool = True) -> OptionValueContainer:
```

Comment:
Previously we piggybacked off `inherit_from_enclosing_scope` as a heuristic to avoid checking deprecations prematurely. Now we're explicit about this.